x-pack/plugins/ingest_manager/server/errors.ts

```diff
@@ -23,6 +43,40 @@ export const getHTTPResponseCode = (error: IngestManagerError): number => {
   return 400; // Bad Request
 };
 
+export const defaultIngestErrorHandler: IngestErrorHandler = async ({
+  error,
+  response,
+}: IngestErrorHandlerParams): Promise<IKibanaResponse> => {
+  const logger = appContextService.getLogger();
+
+  // our "expected" errors
+  if (error instanceof IngestManagerError) {
+    // only log the message
+    logger.error(error.message);
+    return response.customError({
+      statusCode: getHTTPResponseCode(error),
+      body: { message: error.message },
+    });
+  }
+
+  // handle any older Boom-based errors or the few places our app uses them
+  if (isBoom(error)) {
+    // only log the message
+    logger.error(error.output.payload.message);
+    return response.customError({
+      statusCode: error.output.statusCode,
+      body: { message: error.output.payload.message },
+    });
+  }
+
+  // not sure what type of error this is. log as much as possible
+  logger.error(error);
```

Comment:
At this point it might be useful to also log the stack trace, as the `500` passed to `customError` will result in an unhelpful stack trace from platform code.

Reply:
I thought this got us the same as the other two lines (or at least more than the Hapi handler error). I was using it when debugging https://github.com/elastic/kibana/issues/75862 and other stuff

I'll confirm, but I can also leave the original two lines and we can iterate later. I'm most interested in getting more handlers using the default behavior

Reply:
@skh I'm working on tests, but here's my quick local check.


Requesting a missing package will cause a 404 from the registry which becomes a `RegistryResponseError` & 502
```
curl --user elastic:changeme http://localhost:5601/api/ingest_manager/epm/packages/system-9.9.9 | jq
{
  "statusCode": 502,
  "error": "Bad Gateway",
  "message": "'404 Not Found' error response from package registry at http://localhost:8080/package/system/9.9.9/"
}
```

Kibana logs
```
server    log   [12:45:06.493] [error][ingestManager][plugins] '404 Not Found' error response from package registry at http://localhost:8080/package/system/9.9.9/
```

Local mod to cause an error in a service
```diff
diff --git a/x-pack/plugins/ingest_manager/server/services/epm/packages/get.ts b/x-pack/plugins/ingest_manager/server/services/epm/packages/get.ts
index c4232247cc4..eb31e7a6fdb 100644
--- a/x-pack/plugins/ingest_manager/server/services/epm/packages/get.ts
+++ b/x-pack/plugins/ingest_manager/server/services/epm/packages/get.ts
@@ -124,6 +124,8 @@ export async function getInstallationObject(options: {
   pkgName: string;
 }) {
   const { savedObjectsClient, pkgName } = options;
+  // @ts-expect-error
+  errorInsideService();
   return savedObjectsClient
     .get<Installation>(PACKAGES_SAVED_OBJECT_TYPE, pkgName)
     .catch((e) => undefined);
```

same request
```
curl --user elastic:changeme http://localhost:5601/api/ingest_manager/epm/packages/system-9.9.9 | jq
{
  "statusCode": 500,
  "error": "Internal Server Error",
  "message": "errorInsideService is not defined"
}
```

Kibana logs show stack
```
server    log   [12:41:17.260] [error][ingestManager][plugins] ReferenceError: errorInsideService is not defined
    at getInstallationObject (/Users/jfsiii/work/kibana/x-pack/plugins/ingest_manager/server/services/epm/packages/get.ts:128:3)
    at getPackageInfo (/Users/jfsiii/work/kibana/x-pack/plugins/ingest_manager/server/services/epm/packages/get.ts:106:5)
    at handler (/Users/jfsiii/work/kibana/x-pack/plugins/ingest_manager/server/routes/epm/handlers.ts:121:23)
    at /Users/jfsiii/work/kibana/src/core/utils/context.ts:248:14
    at process._tickCallback (internal/process/next_tick.js:68:7)
server   error  [12:41:17.202]  Error: Internal Server Error
    at HapiResponseAdapter.toError (/Users/jfsiii/work/kibana/src/core/server/http/router/response_adapter.ts:129:19)
    at HapiResponseAdapter.toHapiResponse (/Users/jfsiii/work/kibana/src/core/server/http/router/response_adapter.ts:79:19)
    at HapiResponseAdapter.handle (/Users/jfsiii/work/kibana/src/core/server/http/router/response_adapter.ts:74:17)
    at Router.handle (/Users/jfsiii/work/kibana/src/core/server/http/router/router.ts:273:34)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

Reply:
@jfsiii You're right, apologies. I should have run this before commenting.

Reply:
@skh no worries. Good eyes all the same. Take a look at the tests when you get a moment